directions/index.js

```diff
@@ -0,0 +1,147 @@
+import { createClient, createRequest, defaultExchanges, subscriptionExchange } from '@urql/core';
+import { pipe, subscribe } from 'wonka';
+import { SubscriptionClient } from 'subscriptions-transport-ws';
+import { createRoute, routeUpdate } from './queries.js';
+import { drawRoute } from './map.js';
+import * as mapboxPolyline from '@mapbox/polyline';
+import { getDurationString } from '../utils';
+import { createInstructions } from './directions';
+
+/**
+ * Example application of how to build a route with the Chargetrip API.
+ * Please have a look to Readme file in this repo for more details.
+ *
+ * For the purpose of this example we use urgl - lightweights GraphQL client.
+ * To establish a connection with Chargetrip GraphQL API you need to have an API key.
+ * The key in this example is a public one and gives access only to a part of our extensive database.
+ * You need a registered `x-client-id` to access the full database.
+ * Read more about an authorisation in our documentation (https://docs.chargetrip.com/#authorisation).
+ */
+const headers = {
+  'x-client-id': '5e8c22366f9c5f23ab0eff39',
+};
+
+const subscriptionClient = new SubscriptionClient('wss://api.chargetrip.io/graphql', {
+  reconnect: true,
+  connectionParams: headers,
+});
+
+// Add event listeners for 'click' on the tabs.
```

Comment:
Can we move UI code to the bottom of file, to keep the structure clean? Meaning to keep the client initialization first and fetch after, and then the UI part :) It helps to ready the code.